dali/test/python/auto_aug/test_select.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import numpy as np
+from nose2.tools import params
+
+from nvidia.dali import fn, pipeline_def, types
+from nvidia.dali.pipeline import experimental
+from nvidia.dali.auto_aug.core import augmentation, select
+
+from test_utils import get_dali_extra_path, check_batch
+
+data_root = get_dali_extra_path()
+images_dir = os.path.join(data_root, 'db', 'single', 'jpeg')
+
+
+def sample_info(cb):
+
+    def idx_in_batch_cb(sample_info):
+        return np.array(cb(sample_info), dtype=np.int32)
+
+    return fn.external_source(idx_in_batch_cb, batch=False)
```

Comment:
It would be nice to have the op for that at some point. Just a general, unrelated comment.